operator token login_hint format (#218)
camaraproject/IdentityAndConsentManagement: documentation/CAMARA-Security-Interoperability.md

This CAMARA document clarifies the values used in login_hint in the following way:

  * **_tel_**

    For phone numbers. The `login_hint` must be a tel URI as defined in [RFC 3966](https://www.rfc-editor.org/info/rfc3966) for global phone numbers without visual separators in [E.164](https://www.itu.int/rec/T-REC-E.164-201011-I/en) format. For example, `tel:+34666666666`.

  * **_ipport_**

    For IPv4 and IPv6 addresses, that can optionally include a port. For example, `ipport:80.90.34.2:16790`, `ipport:80.90.34.2`, `ipport:[2001:db8::1]:8080` or `ipport:[2001:db8::1]`.

  * **_ot_**

    For operator tokens as defined by [GSMA TS.43](https://www.gsma.com/newsroom/gsma_resources/ts-43-service-entitlement-configuration/) and [GSMA ASAC](https://www.gsma.com/newsroom/gsma_resources/asac-01-v1-0/).
    TS.43 does not specify the format of the operator token and it therefore might contain characters that are not url-safe.

>     This document specifies that the API consumer must encode the operator token using [base64url](https://www.rfc-editor.org/rfc/rfc7515.html#appendix-C) encoding to make it urlsafe.

Reviewer: ```suggestion
    The API consumer MUST encode the operator token using [base64url](https://www.rfc-editor.org/rfc/rfc7515.html#appendix-C) encoding to make it URL safe.
```

    For IPv4 and IPv6 addresses, that can optionally include a port. For example, `ipport:80.90.34.2:16790`, `ipport:80.90.34.2`, `ipport:[2001:db8::1]:8080` or `ipport:[2001:db8::1]`.

  * **_ot_**

    For operator tokens as defined by [GSMA TS.43](https://www.gsma.com/newsroom/gsma_resources/ts-43-service-entitlement-configuration/) and [GSMA ASAC](https://www.gsma.com/newsroom/gsma_resources/asac-01-v1-0/).
    TS.43 does not specify the format of the operator token and it therefore might contain characters that are not url-safe.
    This document specifies that the API consumer must encode the operator token using [base64url](https://www.rfc-editor.org/rfc/rfc7515.html#appendix-C) encoding to make it urlsafe.

    The following is a non-normative example of an OIDC authorization POST request with an ot-login_hint:

```text
    POST /authorize HTTP/1.1
    Host: server.example.com
    Content-Type: application/x-www-form-urlencoded

    response_type=code
      &scope=openid%20dpv:FraudPreventionAndDetection%20sim-swap:check%20sim-swap:retrieve-date
      &client_id=s6BhdRkqt3
      &redirect_uri=https%3A%2F%2Fclient.example.org%2Fcb
      &login_hint="ot:ZXhhbXBsZQ"

> ```

Author: Both other formats have an example

> ```

Reviewer: You can't see the difference? The other examples are only examples of the `login_hint` formatting itself. But for `ot:`, you give a full request example, and a controversial one at that. Why not restrict the example to `ot:ZXhhbXBsZQ`?

> ```

Author: Ok, your suggestion is the removing the example instead of using "ot:ZXhhbXBsZQ". I'll change the example to ot:ZXhhbXBsZQ.
